iXBRLViewerPlugin/viewer/src/js/chart.js

```diff
@@ -18,222 +18,224 @@ import { AspectSet } from './aspect.js';
-}
-
-IXBRLChart.prototype = Object.create(Dialog.prototype);
-
-IXBRLChart.prototype._multiplierDescription = function(m) {
-    var desc = {
-        0: "",
-        3: "'000s",
-        6: "millions",
-        9: "billions",
-    };
-    return desc[m];
-}
-
-IXBRLChart.prototype._chooseMultiplier = function(facts) {
-    var max = 0;
-    $.each(facts, function (i, f) {
-        var v = Number(f.value());
-        if(v > max) {
-            max = v;
-        }
-    });
-    var scale = 0;
-    while (max > 1000 && scale < 9) {
-        max = max / 1000;
-        scale += 3; 
-    } 
-    return scale;
-}
-
-IXBRLChart.prototype.dataSetColour = function(i) {
-    return [
-        '#66cc00',
-        '#0094ff',
-        '#fbad17'
-    ][i];
-}
-
-IXBRLChart.prototype.addAspect = function(a) {
-    this._analyseDims.push(a);
-    this._showAnalyseDimensionChart();
-}
+export class IXBRLChart extends Dialog {
+    constructor() {
+        super(".dialog.chart");
+    }
 
-IXBRLChart.prototype.removeAspect = function(a) {
-    var newDims = [];
-    $.each(this._analyseDims, function (i,d) { if (d != a) { newDims.push(d) }});
-    this._analyseDims = newDims;
-    this._showAnalyseDimensionChart();
-}
+    _multiplierDescription(m) {
+        const desc = {
+            0: "",
+            3: "'000s",
+            6: "millions",
+            9: "billions",
+        };
+        return desc[m];
+    }
 
-IXBRLChart.prototype.analyseDimension = function(fact, dims) {
-    this._analyseFact = fact;
-    this._analyseDims = dims;
-    this._showAnalyseDimensionChart();
-}
+    _chooseMultiplier(facts) {
+        let max = 0;
+        $.each(facts, function (i, f) {
+            var v = Number(f.value());
+            if(v > max) {
+                max = v;
+            }
+        });
+        let scale = 0;
+        while (max > 1000 && scale < 9) {
+            max = max / 1000;
+            scale += 3; 
+        } 
+        return scale;
+    }
 
-/*
- * Create a bar chart show fact values broken down along up to two dimensions
- */
-IXBRLChart.prototype._showAnalyseDimensionChart = function() {
-    var fact = this._analyseFact;
-    var dims = this._analyseDims;
-    var co = this;
-    var c = this.node;
-    $("canvas",c).remove();
-    $("<canvas>").appendTo($(".chart-container",c));
-    this.show();
-
-    /* Find all facts that are aligned with the current fact, except for the
-     * two dimensions that we're breaking down by */
-    var covered = {};
-    if (dims[0]) {
-        covered[dims[0]] = null;
+    dataSetColour(i) {
+        return [
+            '#66cc00',
+            '#0094ff',
+            '#fbad17'
+        ][i];
     }
-    if (dims[1]) {
-        covered[dims[1]] = null;
+
+    addAspect(a) {
+        this._analyseDims.push(a);
+        this._showAnalyseDimensionChart();
     }
 
-    var facts = fact.report().deduplicate(fact.report().getAlignedFacts(fact, covered));
+    removeAspect(a) {
+        var newDims = [];
```

Comment:
nit: some vars still hanging around in this file, as well.

Reply:
Fixed in 7dbd3fbe8f86d74c2f119b0b2e8a78aa4b4420c3